polkadot/xcm/src/v4/mod.rs

```diff
@@ -1314,8 +1313,8 @@ impl<Call: Decode + GetDispatchInfo> TryFrom<NewInstruction<Call>> for Instructi
 			HrmpChannelAccepted { recipient } => Self::HrmpChannelAccepted { recipient },
 			HrmpChannelClosing { initiator, sender, recipient } =>
 				Self::HrmpChannelClosing { initiator, sender, recipient },
-			Transact { origin_kind, mut call } => {
-				let require_weight_at_most = call.take_decoded()?.get_dispatch_info().call_weight;
+			Transact { origin_kind, call, fallback_max_weight } => {
+				let require_weight_at_most = fallback_max_weight.unwrap_or(Weight::MAX);
```

Comment:
This looks incorrect, shouldn't we use `fallback_max_weight` only when `call.take_decoded()?.get_dispatch_info()` is unavailable? Basically only when `call` is `()`?

IIUC with the current code, any XCMv5 Transact with `fallback_max_weight: None` will be converted to a XCMv4 Transact with `require_weight_at_most = Weight::MAX` even for those messages where call weight is available.

Reply:
Messages only get converted on the sender, so `call.take_decoded()?.get_dispatch_info()` is never going to be called in the receiver and work.

We do convert "upwards" in the receiver, but we don't convert "downwards" so this would never be called.

Reply:
What about on the local chain converting a V5 transact (of a local call) to a V4 transact of same local call.

I don’t have an example in mind where this would happen, but we could easily support it, so I’m proposing we use the info if we have it. It’s defensive programming and doesn’t cost us anything.

Reply:
Sure, will add it back